2009-02-13  Michael Tautschnig  <mt@debian.org>

	* lib/get-config-dir-svn: Make username in svn:// config url optional (thanks
		Alexander Fisher) (closes: #441436)
Index: trunk/lib/get-config-dir-svn
===================================================================
--- trunk.orig/lib/get-config-dir-svn
+++ trunk/lib/get-config-dir-svn	
@@ -12,7 +12,11 @@
 # matched string: "svn://user@host/svnpath"
 protocol=$(expr match "$FAI_CONFIG_SRC" '\([^:]*\)://')
 username=$(expr match "$FAI_CONFIG_SRC" '[^:]*://\([^@]*\)@')
-svnpath=$(expr match "$FAI_CONFIG_SRC" '[^:]*://[^@]\+@\([^[:space:]]\+\)')
+if [ -n "$username" ] ; then
+  svnpath=$(expr match "$FAI_CONFIG_SRC" '[^:]*://[^@]\+@\([^[:space:]]\+\)')
+else
+  svnpath=$(expr match "$FAI_CONFIG_SRC" '[^:]*://\([^[:space:]]\+\)')
+fi
 
 case $protocol in
 	svn)
@@ -28,9 +32,11 @@
 		svnurl="https://$svnpath"
 		;;
 	svn+ssh)
-		svnurl="svn+ssh://$username@$svnpath"
-		# this is a bit hackish: if $username is empty, the path gets corrected below... 
-		# explaination: svn co -username foo svn+ssh://bar doesnt work as svn co svn+ssh://foo@bar
+    if [ -n "$username" ] ; then
+		  svnurl="svn+ssh://$username@$svnpath"
+    else
+      svnurl=$FAI_CONFIG_SRC
+    fi
 		;;
 	*)
 		echo "get-config-dir-svn: protocol $protocol not implemented"
@@ -39,7 +45,6 @@
 esac
 		
 [ -n "$username" ] && user="--username $username"
-[ -z "$username" ] && svnurl=$FAI_CONFIG_SRC
 
 if [ -d "$FAI/.svn" ] ; then
    echo "Updating SVN in $FAI"
